pkg/gencred/cmd/main.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	"k8s.io/test-infra/pkg/gencred"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// defaultContextName is the default context name.
+	defaultContextName = "build"
+	// defaultConfigFileName is the default kubeconfig filename.
+	defaultConfigFileName = "kubeconfig.yaml"
+)
+
+// Command-line flags
+var context string
+var name string
+var output string
+var certificate bool
+var serviceaccount bool
+var overwrite bool
+
+// validateFlags validates the command-line flags.
+func validateFlags() {
+	var err error
+
+	if len(name) == 0 {
+		printErrAndExit("-n, --name option is required.", 1)
+	}
+
+	output, err = filepath.Abs(output)
```

Comment:
I would expect this tool to print to stdout when `-o` is unspecified (do we really need the -o option?)

Reply:
Having an `-o, --output` is definitely canonical for cli tools (e.g. `curl`), and I think it is intuitive. I do like the idea of *default* to stdout.

Reply:
Now *default* to stdout.